src/generate_builtins.jl

```diff
@@ -138,11 +138,24 @@ function maybe_evaluate_builtin(frame, call_expr, expand::Bool)
     $head f === Core._apply
         argswrapped = getargs(args, frame)
         if !expand
-            return Some{Any}(Core._apply(getargs(args, frame)...))
+            return Some{Any}(Core._apply(argswrapped...))
         end
         argsflat = Base.append_any((argswrapped[1],), argswrapped[2:end]...)
         new_expr = Expr(:call, map(x->isa(x, Symbol) || isa(x, Expr) || isa(x, QuoteNode) ? QuoteNode(x) : x, argsflat)...)
         return new_expr
+""")
+            continue
+        elseif f === Core.invoke
+            print(io, 
+"""
+    $head f === invoke
+            argswrapped = getargs(args, frame)
+            if !expand
+                return Some{Any}(invoke(argswrapped...))
+            end
+            new_f = which(argswrapped[1], argswrapped[2])
```

Comment:
Since lowered AST shouldn't return methods as `Expr` args, I think this is probably safe. I worry a bit, though, about what might happen if we want to add more aggressive optimization passes; if we suddenly get to a point where we are stashing more objects in Exprs, I think we'll run into trouble.

Maybe best to wrap it in some novel type? Or could we make it a "typical :call expr" like above for `Core._apply`?